v0/destinations/blueshift/transform.js

```diff
@@ -0,0 +1,166 @@
+  getValueFromMessage,
+  getFieldValueFromMessage,
+  removeUndefinedAndNullValues,
+  defaultPutRequestConfig
+} = require("../../util");
+
+const {
+  MAPPING_CONFIG,
+  CONFIG_CATEGORIES,
+  BASE_URL_EU,
+  BASE_URL
+} = require("./config");
+
+function checkValidEventName(str) {
+  if (
+    str.indexOf(".") !== -1 ||
+    str.indexOf(" ") !== -1 ||
+    /[0-9]/.test(str) ||
+    str.length > 64
+  )
+    return true;
+  return false;
+}
+
+const trackResponseBuilder = async (message, category, { Config }) => {
+  let event = getValueFromMessage(message, "event");
+  if (!event) {
+    throw new CustomError(
+      "[Blueshift] property:: event is required for track call",
+      400
+    );
+  }
+  event = event.trim().toLowerCase();
+  if (checkValidEventName(event)) {
+    throw new CustomError(
+      "[Blueshift] Event name doesn't contain period(.), whitespace, numeric value and contains not more than 64 characters",
+      400
+    );
+  }
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+
+  if (!payload) {
+    // fail-safety for developer error
+    throw new CustomError(ErrorMessage.FailedToConstructPayload, 400);
+  }
+  const response = defaultRequestConfig();
+
+  if (Config.datacenterEU) {
+    response.endpoint = `${BASE_URL_EU}/api/v1/event`;
+  } else {
+    response.endpoint = `${BASE_URL}/api/v1/event`;
+  }
+  response.method = defaultPostRequestConfig.requestMethod;
+  const basicAuth = Buffer.from(Config.eventApiKey).toString("base64");
+  response.headers = {
+    Authorization: `Basic ${basicAuth}`,
+    "Content-Type": "application/json"
+  };
+  response.body.JSON = payload;
+  return response;
+};
+
+const identifyResponseBuilder = async (message, category, { Config }) => {
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+
+  if (!payload) {
+    // fail-safety for developer error
+    throw new CustomError(ErrorMessage.FailedToConstructPayload, 400);
+  }
+  const response = defaultRequestConfig();
+
+  if (Config.datacenterEU) {
+    response.endpoint = `${BASE_URL_EU}/api/v1/customers`;
+  } else {
+    response.endpoint = `${BASE_URL}/api/v1/customers`;
+  }
+  response.method = defaultPostRequestConfig.requestMethod;
+  const basicAuth = Buffer.from(Config.usersApiKey).toString("base64");
+  response.headers = {
+    Authorization: `Basic ${basicAuth}`,
+    "Content-Type": "application/json"
+  };
+  response.body.JSON = payload;
+  return response;
+};
+
+const groupResponseBuilder = async (message, category, { Config }) => {
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+
+  if (!payload) {
+    // fail-safety for developer error
+    throw new CustomError(ErrorMessage.FailedToConstructPayload, 400);
+  }
+  const response = defaultRequestConfig();
+
+  const customer = removeUndefinedAndNullValues({
+    id: getFieldValueFromMessage(message, "userId"),
+    email: getFieldValueFromMessage(message, "email")
+  });
```

Comment:
Do we really need to bind to a customer variable? Can we use `const id` and `const email` ?